scalikejdbc-streams/src/main/scala/scalikejdbc/streams/DatabasePublisherProvider.scala

```diff
@@ -0,0 +1,48 @@
+package scalikejdbc.streams
+
+import scalikejdbc.{ ConnectionPool, DB, SettingsProvider, WithExtractor }
+
+import scala.concurrent.ExecutionContext
+
+/**
+ * Provider of DatabasePublisher.
+ */
+final class DatabasePublisherProvider private[streams] (
+    private[streams] val connectionPoolName: Any = ConnectionPool.DEFAULT_NAME,
+    private[streams] val settings: SettingsProvider = SettingsProvider.default,
+    private[streams] val sessionModification: SessionModification = SessionModification.default
+) {
+
+  /**
+   * Returns a provider instance providing a DatabasePublisher
+   * that uses the DB session modified with SessionModification.
+   */
+  def withSessionModification(sessionModification: SessionModification): DatabasePublisherProvider = {
```

Comment:
Introducing this class only for having this method is not so good to me. Really needed?